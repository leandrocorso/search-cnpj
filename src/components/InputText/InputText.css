.form-group {
	padding: 1rem;
	background-color: #fff;
}

.form-group label {
    font-size: 80%;
    color: #999;
}

.form-group .input-group {
	width: 100%;
	display: flex;
	align-items: baseline;
	justify-content: space-between;
	margin-top: 1rem;
}

.form-group .input-group .inputTest {
	font-size: 150%;
}

.form-group .icon {
	width: 1rem;
	height: 1rem;
	border-radius: 50%;
	border: 2px solid #fff;
}

.form-group .icon.success {
	fill: #fff;
	border-color: #85cd1a;
	background-color: #85cd1a;
}

.form-group .icon.danger {
	fill: #fff;
	border-color: #f66;
	background-color: #f66;
}