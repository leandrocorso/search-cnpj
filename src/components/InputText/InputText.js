import React, { Component } from 'react';
import MaskedInput from 'react-text-mask';
import styled from 'styled-components';

import { Icon } from '../../components';

const FormGroup = styled.div`
	height: 4.5rem;
	padding: 1rem;
	background-color: ${p => p.theme.themeDefault.colors.c4};

	label {
		font-size: 80%;
		color: ${p => p.theme.themeDefault.colors.c7};
	}
`;

const InputGroup = styled.div`
	width: 100%;
	display: flex;
	align-items: baseline;
	justify-content: space-between;
`;

const CheckIcon = styled(Icon)`
	width: .8rem;
	height: .8rem;
	fill: ${ p => p.theme.themeDefault.colors.c3 };
`;

class InputText extends Component {
	constructor(props) {
		super(props);
		this.state = { value: '' }
	}
  
	handleChange = value => {
		const { onChange } = this.props;

		this.setState({ value }, () => { 
			if (onChange) onChange(value);
		});
	};

  render() {
    const { name, label, isValid, ...rest } = this.props;

    return (
    	<FormGroup>
    		<label htmlFor={name}>{label}</label>
			<InputGroup>
				<MaskedInput
					{...rest}
					id={name}
					name={name}
					onChange={e => this.handleChange(e.target.value)}
					value={this.state.value}
					className="inputTest"
					/>
				{ isValid && <CheckIcon src="check-alt" /> }
			</InputGroup>
    	</FormGroup>
    	);
	}
}

export default InputText;
