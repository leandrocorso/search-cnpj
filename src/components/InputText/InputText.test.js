import React from 'react';
import { shallow } from 'enzyme';

import InputText from './InputText';

describe('Testing InputText component', () => {
    
    const wrapper = shallow(<InputText />);
    
    test('InputText exists', () => {
        expect(wrapper.find('input'));
    });

    test('InputText have a name', () => {
        expect(wrapper.find('.inputTest')).toHaveProp('name');
    });

    test('Label element exists', () => {
        expect(wrapper.find('label'));
    });

    test('Label have reference to input text', () => {
        expect(wrapper.find('label')).toHaveProp('htmlFor');
    });  

    test('Should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

})