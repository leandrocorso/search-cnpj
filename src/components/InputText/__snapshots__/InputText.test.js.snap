// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing InputText component Should render correctly 1`] = `
<styled.div>
  <label />
  <styled.div>
    <t
      className="inputTest"
      onChange={[Function]}
      render={[Function]}
      value=""
    />
  </styled.div>
</styled.div>
`;
